classes/gateways/class.pmprogateway_stripe.php

```diff
@@ -4317,16 +4317,26 @@ static function refund( $success, $order ) {
 			$client = new Stripe_Client( $secretkey );
 			$refund = $client->refunds->create( [
 				'charge' => $transaction_id,
-			] );
+			] );			
 
-			$success = true;
+			//Make sure we're refunding an order that was successful
+			if ( $refund->status == 'succeeded' ) {
+
+				$order->status = 'refunded';	
+
+				$success = true;
 			
-			global $current_user;
+				global $current_user;
 
-			$order->notes = $order->notes.' '.sprintf( __('Order successfully refunded on %1$s for transaction  ID %2$s by %3$s', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $transaction_id, $current_user->display_name );	
+				$order->notes = trim( $order->notes.' '.sprintf( __('Order successfully refunded on %1$s for transaction ID %2$s by %3$s', 'paid-memberships-pro' ), date_i18n('Y-m-d H:i:s'), $transaction_id, $current_user->display_name ) );	
 
-			if ( $refund->status == 'succeeded' ) {
-				$order->status = 'refunded';					
+				//send an email to the member
+				$myemail = new PMProEmail();
+				$myemail->sendRefundedEmail( $current_user );
+
+				//send an email to the admin
+				$myemail = new PMProEmail();
+				$myemail->sendRefundedAdminEmail( $current_user, $order->membership_id );				
 			} else {
 				$order->notes = trim( $order->notes ) .' '. __('An error occured while attempting to process this refund.', 'paid-memberships-pro' );
```

Comment:
The `trim()` should wrap the whole new note and not just the previous note.

```suggestion
				$order->notes = trim( $order->notes . ' ' . __('An error occured while attempting to process this refund.', 'paid-memberships-pro' ) );
```